lib/helper/Playwright.js

```diff
@@ -1603,15 +1601,9 @@ class Playwright extends Helper {
     const matcher = await this.context;
     let waiter;
     const context = await this._getContext();
-    if (locator.isCSS()) {
-      const valueFn = function (locator, value) {
-        const els = document.querySelectorAll(locator);
-        if (!els || els.length === 0) {
-          return false;
-        }
-        return Array.prototype.filter.call(els, el => (el.value || '').indexOf(value) !== -1).length > 0;
-      };
-      waiter = context.waitForFunction(valueFn, { timeout: waitTimeout }, locator.value, value);
+    if (!locator.isXPath()) {
+      // uses a custom selector engine for finding value properties on elements
+      waiter = context.waitForSelector(`${locator.type}=${locator.value} >> __value=${value}`, { timeout: waitTimeout });
```

Comment:
Wrote a basic custom engine to find elements based on their value which can be put into `waitForSelector` calls. I'm discussing other options here https://github.com/microsoft/playwright/issues/1444#issuecomment-602688759